src/libp2p_ipc/libp2p_ipc.capnp

```diff
@@ -98,282 +73,319 @@ struct ValidationMessage {
+      getListeningAddrs @3 :Libp2pHelperInterface.GetListeningAddrs.Response;
+      beginAdvertising @4 :Libp2pHelperInterface.BeginAdvertising.Response;
+      addPeer @5 :Libp2pHelperInterface.AddPeer.Response;
+      listPeers @6 :Libp2pHelperInterface.ListPeers.Response;
+      generateKeypair @7 :Libp2pHelperInterface.GenerateKeypair.Response;
+      publish @8 :Libp2pHelperInterface.Publish.Response;
+      subscribe @9 :Libp2pHelperInterface.Subscribe.Response;
+      unsubscribe @10 :Libp2pHelperInterface.Unsubscribe.Response;
+      addStreamHandler @11 :Libp2pHelperInterface.AddStreamHandler.Response;
+      removeStreamHandler @12 :Libp2pHelperInterface.RemoveStreamHandler.Response;
+      openStream @13 :Libp2pHelperInterface.OpenStream.Response;
+      closeStream @14 :Libp2pHelperInterface.CloseStream.Response;
+      resetStream @15 :Libp2pHelperInterface.ResetStream.Response;
+      sendStream @16 :Libp2pHelperInterface.SendStream.Response;
+      setNodeStatus @17 :Libp2pHelperInterface.SetNodeStatus.Response;
+      getPeerNodeStatus @18 :Libp2pHelperInterface.GetPeerNodeStatus.Response;
+    }
+  }
+
+  struct RpcResponse {
+    header @0 :RpcMessageHeader;
+    union {
+      error @1 :Text;
+      success @2 :RpcResponseSuccess;
+    }
+  }
+
+  struct PushMessage {
+    header @0 :PushMessageHeader;
+
+    # union {
+    validation @1 :Libp2pHelperInterface.Validation;
+    # }
+  }
+
+  struct IncomingMessage {
+    union {
+      rpcRequest @0 :Libp2pHelperInterface.RpcRequest;
+      pushMessage @1 :Libp2pHelperInterface.PushMessage;
+    }
   }
 }
 
 # all messages in the daemon interface are push messages 
 struct DaemonInterface {
   struct PeerConnected {
-    peerId @0 :PushMessage(Text);
+    peerId @0 :PeerId;
   }
 
   struct GossipReceived {
-    msg @0 :PushMessage(GossipMessage);
+    sender @0 :PeerInfo;
+    seenAt @1 :UInt64;
+    expiration @2 :UInt64;
+    subscriptionId @3 :UInt64;
+    validationSeqNumber @4 :UInt64;
+    data @5 :Data;
   }
 
   struct IncomingStream {
-    stream @0 :PushMessage(IncomingStream);
+    id @0 :UInt64;
+    peer @1 :PeerInfo;
+    protocol @2 :Text;
   }
 
   struct StreamLost {
-    streamLost @0 :PushMessage(StreamLost);
+    id @0 :UInt64;
+    reason @1 :Text;
   }
 
   struct StreamComplete {
-    streamId @0 :PushMessage(StreamId);
+    streamId @0 :StreamId;
   }
 
   struct StreamMessageReceived {
-    msg @0 :PushMessage(StreamMessage);
+    msg @0 :StreamMessage;
   }
-}
+
+  struct PushMessage {
+    header @0 :PushMessageHeader;
+
+    union {
+      peerConnected @1 :DaemonInterface.PeerConnected;
+      gossipReceived @2 :DaemonInterface.GossipReceived;
+      incomingStream @3 :DaemonInterface.IncomingStream;
+      streamLost @4 :DaemonInterface.StreamLost;
+      streamComplete @5 :DaemonInterface.StreamComplete;
+      streamMessageReceived @6 :DaemonInterface.StreamMessageReceived;
+    }
+  }
+
+  struct IncomingMessage {
```

Comment:
Let's use just `Message` maybe?

Because it's outgoing for Ocaml code and incoming for Go.